model/ddl.go

```diff
@@ -282,9 +282,20 @@ func (job *Job) Decode(b []byte) error {
 
 // DecodeArgs decodes job args.
 func (job *Job) DecodeArgs(args ...interface{}) error {
-	job.Args = args
-	err := json.Unmarshal(job.RawArgs, &job.Args)
-	return errors.Trace(err)
+	var rawArgs []json.RawMessage
+	if err := json.Unmarshal(job.RawArgs, &rawArgs); err != nil {
+		return err
+	}
+
+	for i := 0; ; i++ {
+		if i == len(rawArgs) || i == len(args) {
+			job.Args = args[:i]
+			return nil
+		}
+		if err := json.Unmarshal(rawArgs[i], args[i]); err != nil {
+			return err
+		}
+	}
```

Comment:
```suggestion
	if len(rawArgs) > len(args) {
		rawArgs = rawArgs[:len(args)]
	}
	for i, rawArg := range rawArgs {
		if err := json.Unmarshal(rawArg, args[i]); err != nil {
			return err
		}
	}
	job.Args = args
	return nil
```

Reply:
We need to ensure if `len(rawArgs) == 0` then `len(job.Args) == 0`.